packages/flutter_tools/test/general.shard/android/gradle_utils_test.dart

```diff
@@ -967,19 +967,21 @@ allprojects {
       JavaAgpTestData(true, javaVersion: '18', agpVersion: '4.2'),
       // Strictly too new AGP versions.
       // *The tests that follow need to be updated* when max supported AGP versions are updated:
-      JavaAgpTestData(false, javaVersion: '24', agpVersion: '8.3'),
-      JavaAgpTestData(false, javaVersion: '20', agpVersion: '8.3'),
-      JavaAgpTestData(false, javaVersion: '17', agpVersion: '8.3'),
+      JavaAgpTestData(false, javaVersion: '24', agpVersion: '8.5'),
+      JavaAgpTestData(false, javaVersion: '20', agpVersion: '8.5'),
+      JavaAgpTestData(false, javaVersion: '17', agpVersion: '8.5'),
       // Java 17 & patch versions compatibility cases
       // *The tests that follow need to be updated* when maxKnownAndSupportedAgpVersion is
       // updated:
-      JavaAgpTestData(false, javaVersion: '17', agpVersion: '8.2'),
+      JavaAgpTestData(false, javaVersion: '17', agpVersion: '8.5'),
       JavaAgpTestData(true, javaVersion: '17', agpVersion: maxKnownAndSupportedAgpVersion),
+      JavaAgpTestData(true, javaVersion: '17', agpVersion: '8.3'),
       JavaAgpTestData(true, javaVersion: '17', agpVersion: '8.1'),
       JavaAgpTestData(true, javaVersion: '17', agpVersion: '8.0'),
       JavaAgpTestData(true, javaVersion: '17', agpVersion: '7.4'),
-      JavaAgpTestData(false, javaVersion: '17.0.3', agpVersion: '8.2'),
+      JavaAgpTestData(false, javaVersion: '17.0.3', agpVersion: '8.5'),
       JavaAgpTestData(true, javaVersion: '17.0.3', agpVersion: maxKnownAndSupportedAgpVersion),
+      JavaAgpTestData(true, javaVersion: '17.0.3', agpVersion: '8.3'),
```

Comment:
Should we add a test for 8.2 here?

Reply:
I've added one, but the tests here will never be exhaustive regardless because it is really only a test of the unique combination of Java `17.0.3` and AGP `8.2`, not a test of AGP `8.2` independently